Modules/Quotas/src/Quotas.Application/Identities/Queries/GetQuotasForIdentity/Handler.cs

```diff
@@ -0,0 +1,80 @@
+﻿using Backbone.BuildingBlocks.Application.Abstractions.Exceptions;
+using Backbone.BuildingBlocks.Application.Abstractions.Infrastructure.UserContext;
+using Backbone.BuildingBlocks.Application.Pagination;
+using Backbone.DevelopmentKit.Identity.ValueObjects;
+using Backbone.Modules.Quotas.Application.DTOs;
+using Backbone.Modules.Quotas.Application.Infrastructure.Persistence.Repository;
+using Backbone.Modules.Quotas.Domain.Aggregates.Identities;
+using Backbone.Modules.Quotas.Domain.Aggregates.Metrics;
+using Backbone.Modules.Quotas.Domain.Metrics;
+using MediatR;
+
+namespace Backbone.Modules.Quotas.Application.Identities.Queries.GetQuotasForIdentity;
+
+public class Handler : IRequestHandler<ListQuotasForIdentityQuery, ListQuotasForIdentityResponse>
+{
+    private readonly IIdentitiesRepository _identitiesRepository;
+    private readonly MetricCalculatorFactory _metricCalculatorFactory;
+    private readonly IdentityAddress _identityAddress;
+
+    public Handler(IUserContext userContext, IIdentitiesRepository identitiesRepository, MetricCalculatorFactory metricCalculatorFactory)
+    {
+        _identityAddress = userContext.GetAddress();
+        _identitiesRepository = identitiesRepository;
+        _metricCalculatorFactory = metricCalculatorFactory;
+    }
+
+    public async Task<ListQuotasForIdentityResponse> Handle(ListQuotasForIdentityQuery request, CancellationToken cancellationToken)
+    {
+        var identity = await _identitiesRepository.Find(_identityAddress, cancellationToken) ?? throw new NotFoundException(nameof(Identity));
+
+        var individualQuotasForIdentityTasks = identity.IndividualQuotas
+            .Select(async q =>
+            {
+                var calculator = _metricCalculatorFactory.CreateFor(q.MetricKey);
+                var usage = await calculator.CalculateUsage(q.Period.CalculateBegin(), q.Period.CalculateEnd(), identity.Address, cancellationToken);
+
+                return new QuotaDTO(
+                    q,
+                    new MetricDTO(new Metric(q.MetricKey, q.MetricKey.ToString()!)),
```

Comment:
Why are you just setting the display name of the Metric to the same value as the key?

You can get the metrics (including their display name) from the MetricsRepository. So after you retrieved the Identity, you should get all Metrics, and then use the in memory list to get the metric you're looking for for each quota.